src/expr.h

```diff
@@ -85,6 +99,8 @@ class TypeCastExpr : public Expr {
         bool is_implicit;
```

Comment:
What is_implicit flag means?
Who decides that it's implicit or explicit? IR node should indicate if this conversion can be implicit or cannot. I.e. if it's ommitted - will the program change hte behavior.
Anyway, just make a comment about who decides if this is implicit or not.